src/systems/filecoin_mining/storage_mining/storage_miner_actor.id

```diff
@@ -9,26 +9,51 @@ type SectorCommitment struct {}
 
 type StorageMinerActor struct {
     // CollateralVault CollateralVault
-    // Sectors {UInt:MinedSector} // sectors and faults, proving period ends
-    // CommitSector(CommD Commitment, CommR Commitment, proof SealProof, SectorSize UVarint)
-    // ProvingPeriodEnd   Epoch
 
-    // // contains mostly static info about this miner
-    // info &MinerInfo
+    // contains mostly static info about this miner
+    info &MinerInfo
+
+    // TODO: add the DePledgeCollateral
+    // TODO: add the Sectors
+    // TODO sectors []SectorInfo CID to AMT
+    // TODO provingSet []SectorInfo CID to AMT
+
+    // TODO: should we keep the same structure as before or putting all of that into a struct?
+    // TODO currentFaultSet BitField
+    // TODO nextFaultSet BitField
+    // TODO nextDoneSet BitField
+    // TODO do we need a `slashedAt`?
+    // TODO owedStorageCollateral
+    // TODO ProvingPeriodEnd   Epoch
 
-    UpdateDoneSet()
-    UpdateSectorStatus(sectorStateSets sector.SectorStateSets)
-    SubmitPoSt(
-        postProof        base_mining.PoStProof
-        sectorStateSets  sector.SectorStateSets
-    )
-    computeProvingPeriodEndSectorState()
     CommitSector(
         seed            Seed
         sector          sector.SectorID
-        sealCommitment  base_mining.SealCommitment
+        sealCommitment  base_mining.SealCommitment  // TODO: it's more expensive to do a struct
```

Comment:
- This is almost certainly going to remain as a struct. I understand the additional cost.
- `base_mining.SealCommitment` looks wrong. I believe this needs to be `sector.OnChainSealVerifyInfo`

Reply:
👍